### DISCUSSION TITLE
[Time series echart] Need a threshold to show value labels.

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18524] [Ideas]
**Is your feature request related to a problem? Please describe.**
A new feature is available in master, where user can select to show only total or all values for a stacked series. When showing all values, we see that the values are very cluttered when many series have close/similar values.

Sample:
![image](https://user-images.githubusercontent.com/74634977/139378228-8e69cc7a-eb69-4048-ae85-52c1633cd617.png)

**Describe the solution you'd like**
A good solution for this is in Pie chart, where the user can input the minimum threshold in percentage to decide which labels should be seen in chart.
![image](https://user-images.githubusercontent.com/74634977/139377720-8b965129-9c0d-4c04-ab3d-c72092a853eb.png)

I guess that if such threshold value is added by user in Time-series chart, then need to calculate % value of each series at each point on x-axis to decide which labels to show on every point of x-axis.

**Describe alternatives you've considered**
Another solution would be to set a fixed threshold from code, but I guess giving the option to user would be a better idea.

**Additional context**

**Comment by villebro** [comment]
This is an excellent suggestion! @junlincc @stephenLYZ let's consider how this could best be implemented.

**Comment by stephenLYZ** [comment]
Thanks for the suggestion! And the idea that calculating % value of each series are also very valuable.

**Comment by junlincc** [comment]
Absolutely agree! thanks for suggesting! @kamalkeshavani-aiinside. 

The change seems to be a good first issue. I'm okay with simply adding a text input field.

**Comment by corbinrobb** [comment]
Do you mind if I give this a try? @junlincc

**Comment by eschutho** [comment]
yes, go for it @corbinrobb! Thank you!

**Comment by corbinrobb** [comment]
No problem! @eschutho I will have a PR up shortly
